lib/maxmind/geoip2/model/abstract.rb

```diff
@@ -0,0 +1,29 @@
+# frozen_string_literal: true
+
+require 'ipaddr'
+
+module MaxMind::GeoIP2::Model
+  # @!visibility private
+  class Abstract
+    def initialize(record)
+      @record = record
+
+      ip = IPAddr.new(
+        format('%s/%d', record['ip_address'], record['prefix_length']),
+      )
```

Comment:
You could also just set the prefix on the `IPAddr` object rather than creating a CIDR-like string.

I wonder if it would be nicer to provide objects rather than strings for `ip_address` and `network`, especially given we are creating them anyway.

Reply:
Using `prefix=`? Unfortunately that's not available in Ruby 2.4. Although we could use `mask` I think.

I'm not sure about returning the objects. For network I'm inclined to say no, because I can't see a way to get the prefix out of it once I create it (in Ruby 2.4 anyway), not even in CIDR form.

Reply:
Ah, ok. Yeah, `IPAddr` is kind of a weird class. I didn't have strong feelings on this.